components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -1720,4 +1724,47 @@ private void handleRequestForANonExistingApp(String appName) throws InvalidOAuth
         }
         throw new InvalidOAuthClientException(message);
     }
+
+    /**
+     * Check whether a client ID unique key constraint exists in the IDN_OAUTH_CONSUMER_APPS table.
+     * This is required to check compatibility with client ID tenant unification.
+     *
+     * @return true if the unique key constraint exists, false otherwise.
+     */
+    public boolean isClientIDUniqueConstraintExistsInConsumerAppsTable()
+            throws IdentityOAuth2Exception {
+
+        String sqlQuery = SQLQueries.OAuthAppDAOSQLQueries.CHECK_CONSUMER_KEY_CONSTRAINT_ON_CONSUMER_APPS_TABLE;
+        String tableName = CONSUMER_APPS_TABLE_NAME;
+        String keyColumnName = CONSUMER_KEY;
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            DatabaseMetaData metaData = connection.getMetaData();
+
+            if (metaData.getDatabaseProductName().toLowerCase().contains("db2")) {
```

Comment:
There was a util class to check the DB type, please use that

Reply:
Fixed